Dockerfile

```diff
@@ -1,66 +1,134 @@
+    curl -fsSL https://deb.nodesource.com/setup_lts.x | bash - && \
+    apt update && \
+    apt install -y --no-install-recommends nodejs && \
+    apt clean && \
+    rm -rf /var/lib/apt/lists/* && \
+    npm install -g npm && \
+    npm install -g yarn && \
+    npm install -g pnpm 
 
-RUN set -x \
-    # Install build tools
-    && RUSTFLAGS=-Ctarget-feature=-crt-static cargo install wasm-pack cargo-chef \
-    && npm install rollup \
-    && rustup target add wasm32-unknown-unknown
+#######################################################
+### Only enable if building non-native architecture ###
+#######################################################
+## aarch64 build 
+### Set openssl path
+#ENV OPENSSL_LIB_DIR="/usr/lib/aarch64-linux-gnu/"
+#RUN dpkg --add-architecture arm64 && \
+#    apt update && \
+#    apt install -y gcc-aarch64-linux-gnu g++-aarch64-linux-gnu libc6-arm64-cross libc6-dev-arm64-cross libssl-dev:arm64 && \
+#    apt clean && \
+#    rm -rf /var/lib/apt/lists/*
+### add arm64 target
+#RUN rustup target add aarch64-unknown-linux-gnu
 
-# Prepare the dependency list.
-FROM chef AS planner
-COPY . .
-RUN cargo chef prepare --recipe-path /tmp/recipe.json
+## armhf build
+### Set openssl path
+#ENV OPENSSL_LIB_DIR="/usr/lib/arm-linux-gnueabihf/"
+#RUN dpkg --add-architecture armhf && \
+#    apt update && \
+#    apt install -y gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf libc6-armhf-cross libc6-dev-armhf-cross libssl-dev:armhf && \
+#    apt clean && \
+#    rm -rf /var/lib/apt/lists/*
+### add armhf target
+#RUN rustup target add rustup target add armv7-unknown-linux-gnueabihf
 
-# Build dependencies.
-FROM chef AS builder
-COPY --from=planner /tmp/recipe.json recipe.json
-RUN cargo chef cook --release -p lldap_app --target wasm32-unknown-unknown \
-    && cargo chef cook --release -p lldap \
-    && cargo chef cook --release -p migration-tool
 
-# Copy the source and build the app and server.
-COPY --chown=app:app . .
-RUN cargo build --release -p lldap -p migration-tool \
-    # Build the frontend.
-    && ./app/build.sh
+# Install cargo-chef and wasm-pack via cargo [npm dont have arm64 bin]    
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo install cargo-chef wasm-pack
+RUN rustup target add wasm32-unknown-unknown
+# Install rollup
+RUN npm install -g rollup
 
-# Final image
-FROM alpine:3.14
 
+# Prepare dependencies
+FROM builder-base AS planner
 WORKDIR /app
+COPY . .
+RUN cargo chef prepare --recipe-path /tmp/recipe.json
+
+# Build depedencies
+FROM builder-base AS builder
+COPY --from=planner /tmp/recipe.json recipe.json
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p lldap_app --target wasm32-unknown-unknown
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p lldap
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p migration-tool
+COPY . /lldap-src
```

Comment:
why do you move to another directory?
I think in that case the `target/` directory with all the dependencies is not going to follow and you'll be throwing away all the work that you just did.